client/src/component/AddVideo.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from "react";
+
+function Addvideo() {
+  const [videos, setVideos] = useState([]);
+  const [newVideo, setNewVideo] = useState({ title: "", url: "" });
+  const [idCounter, setIdCounter] = useState(1);
+
+  useEffect(() => {
+    fetchVideos();
+  }, []);
+
+  const fetchVideos = async () => {
+    try {
+      const response = await fetch(
+        "https://node-server-full-stack.onrender.com/videos"
+      );
+      if (response.ok) {
+        // const data = await response.json();
+        // setVideos(data);
+      } else {
+        console.error("Failed to fetch videos:", response.statusText);
+      }
+    } catch (error) {
+      console.error("Error fetching videos:", error);
+    }
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setNewVideo({ ...newVideo, [name]: value });
+  };
+
+  const getYouTubeVideoId = (url) => {
+    const videoIdMatch = url.match(/(?:\/|v=)([A-Za-z0-9_-]{11})(?=&|$)/);
+    if (videoIdMatch) {
+      return videoIdMatch[1];
+    }
+    return null;
+  };
+
+  const addVideo = async () => {
+    if (newVideo.title && newVideo.url) {
+      const videoId = getYouTubeVideoId(newVideo.url);
+
+      if (videoId) {
```

Comment:
This code totally works, but as a style thing, some people would write this as:

```js
if (!videoId) {
  alert("Invalid YouTube URL. Please enter a valid YouTube video URL.");
  return;
}

const videoToAdd = ...
...
```

There are three nice ideas here:
1. Your code isn't as indented, so is a bit easier to read.
2. Detecting what constitutes an error (no videoId) and what we do about it (alert, and don't do the rest of the function) is close together. Right now, when we get to the else, it's pretty easy to forget what the `if` condition was, because it's so far away.
3. It allows us to mostly read the function as a straight line - we don't need to jump between lots of ifs and elses - reading the function top-to-bottom, it does things in order, but may stop early, rather than jumping around so much.